vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gatt_server.c

```diff
@@ -391,11 +261,19 @@ ble_uuid_t * prvCopytoESPUUID( BTUuid_t * pxUuid )
     switch( pxUuid->ucType )
     {
         case eBTuuidType16:
-            uuid16 = pvPortCalloc( 1, sizeof( ble_uuid16_t ) );
 
-            if( !uuid16 )
+            if( pUuid == NULL )
             {
-                return NULL;
+                uuid16 = pvPortCalloc( 1, sizeof( ble_uuid16_t ) );
```

Comment:
Its better to always allocate the memory for ble_uuid_t outside the function ? That way, caller need not have to remember to free the memory allocated from inside the function.